src-built-in/components/appLauncher2/src/components/Content.jsx

```diff
@@ -1,37 +1,77 @@
 import React from  'react'
 import AppDefinition from './AppDefinition'
 import FilterSort from './FilterSort'
+import {getStore} from '../stores/LauncherStore'
+import storeActions from '../stores/StoreActions'
+import sortFunctions from '../utils/sort-functions'
 
 export default class Content extends React.Component {
 
 	constructor(props) {
 		super(props)
 		this.state = {
-			filter: {
-				text: ''
-			}
+			sortBy: storeActions.getSortBy(),
+			filterText: storeActions.getSearchText(),
+			folder: storeActions.getActiveFolder()
 		}
+		this.onSort = this.onSort.bind(this)
+		this.onSearch = this.onSearch.bind(this)
+		this.onActiveFolderChanged = this.onActiveFolderChanged.bind(this)
 	}
 
 	filterApps() {
-		const apps = this.props.activeFolder.appDefinitions
-		if (!this.state.filter.text) {
+		const sortFunc =  sortFunctions[this.state.sortBy]
+		const apps = sortFunc(this.state.folder.appDefinitions)
+		if (!this.state.filterText) {
 			return apps
 		}
 		return apps.filter((app) => {
-			return app.friendlyName.indexOf(this.state.filter.text) > -1
+			return app.name.indexOf(this.state.filterText) !== -1 || 
+			app.friendlyName.indexOf(this.state.filterText) !== -1
 		})
 	}
+	onActiveFolderChanged(error, data) {
+		this.setState({
+			folder: storeActions.getActiveFolder()
+		})
+	}
+	
+	onSearch(error, data) {
+		this.setState({
+			filterText: storeActions.getSearchText()
+		})
+	}
+	
+	onSort(error, data) {
+		this.setState({
+			sortBy: data.value
+		})
+	}
+
+	componentWillMount() {
+		getStore().addListener({field: 'activeFolder'}, this.onActiveFolderChanged)
```

Comment:
Let's go ahead and use a variable here.

```
let store = getStore();
store.addListener...
```

Reply:
Is there a good reason for creating unnecessary variables? Im a believer in "unnecessary variables are bad for your code"

Reply:
I'm a believer in the rule of 3. If I see an object being used more than 3 times, I'd like to see a variable. For me, variables really help when I'm reading the code. I see `store = getStore()`, and I can file that away mentally.

Then, when I see store later down in the code, I know it's the return value of getStore.

https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming) for a little context